src/MainWindow.vala

```diff
@@ -69,4 +70,29 @@ public class Installer.MainWindow : Gtk.Dialog {
             stack.set_visible_child_name ("progress-view");
         });
     }
+
+    private string get_os_release () {
+        string os;
+        var file = File.new_for_path("/etc/os-release");
+
+        try {
+            var osrel = new Gee.HashMap<string, string> ();
```

Comment:
If this is the only information we're gonna get from that file, this `HashMap` isn't needed here and we can just assign the OS string in the `while` statement and then `break` after we're done. If that's not the case we should probably make a new class to parse and provide the needed information about the OS to not clutter the `MainWindow` class.